src/controllers/pods.ts

```diff
@@ -81,56 +81,95 @@ export default function (server: Hapi.Server, deps: Injector) {
     return duration
   }
 
-  // TODO: how to add plugin decorate functions to Hapi.Request type
-  async function postPod (request: any, h: Hapi.ResponseToolkit): Promise<PostPodResponse> {
+  async function streamPod (request: any, h: Hapi.ResponseToolkit) {
+    const res = request.raw.res
+    const streamer = setInterval(() => {
+      res.write(' ')
+    }, config.timeout)
+
+    let method
+    if (request.method === 'post') {
+      // throw error if memory usage exceeds available memory
+      const podSpec = manifestParser.manifestToPodSpec(
+        request.payload['manifest'],
+        request.payload['private'] || {}
+      )
+      if (checkIfHostFull(podSpec)) {
+        throw Boom.serverUnavailable('Memory usage exceeded. Send pod request later.')
+      }
+      method = postPod
+    } else if (method === 'put') {
+      method = extendPod
+    } else {
+      log.error('error uploading pod. error=Invalid method')
+      throw Boom.methodNotAllowed('Invalid method.')
+    }
+
     const duration = await chargeForDuration(request)
 
+    try {
+      const result = await method(request, duration)
+      console.log('result acquired')
+      clearInterval(streamer)
+      console.log('interval cleared')
+      res.setHeader('Content-type', 'application/json')
+      console.log('headers set')
+      res.end(JSON.stringify(result))
+      console.log('returned response')
+    } catch (e) {
+      console.log('caught an error at pods')
+      clearInterval(streamer)
+      // res.statusCode = 500
+      res.writeHead(500, 'Internal Serve Error')
```

Comment:
Doesn't Hapi handle returning 500s?  It's better to never explicitly return a 500 and simply re-throw an error if it's not recoverable